### ISSUE TITLE
Improved legibility of the "Get Code" output, part 2.

### ISSUE THREAD

**Issue by Brugman** [WebDevStudios/custom-post-type-ui#761]
What do you think about going from [this](https://gist.github.com/toscani/57761c95305865b6a2ca1998a36a447f) to [this](https://gist.github.com/toscani/690f626818639342d1af22c2753af30d)?
I'd like to write the PR for this. It'll be a little more challenging than #582 but I've probably gotten better too. (:

Reasons for the changes:

The values are aligned horizontally. Get Code is for taking your code out of CPTUI and managing it by hand from now on. In my opinion the alignment helps you lots visually.

Double quotes to single quotes. I believe HTML uses double quotes by default, and PHP uses single quotes. I have no source on this, just feels like that's most common. In this output there's no need for double quotes, so I would default to singles.

Replaced the functions with closures. Closures got added in PHP 5.3 (2009, dead in 2014). If you only reference a function once then you can use a closure instead. The added benefit is not having to name a function. Naming isn't fun, and if the name isn't unique you get an error.

Replaced `array()` with `[]`. `[]` got added in PHP 5.4 (2012, dead in 2015). It's just shorter, cleaner, nicer.

All* array items are on their own line. This makes all lists vertical, for easier reading/finding. Good text editors let us move/cut/paste/duplicate lines. If the arrays are written like this you can manipulate them much easier.

I think that's all I changed. Looking forward to your thoughts.

**Comment by tw2113** [comment]
Thoughts incoming.

> The values are aligned horizontally. Get Code is for taking your code out of CPTUI and managing it by hand from now on. In my opinion the alignment helps you lots visually.

I agree that well aligned code is good code, but I think you'll find it's harder to achieve alignment perfection in this type of situation than you realize. While it won't be universal by any means, people will hopefully be pasting this type of information in editors that will auto-cleanup this detail as well. Here's to hoping. Admittedly I'm less worried about code alignment for someone moving away, than I am their post types and taxonomies still getting registered/found/etc. I want that part of the migration seamless.

> Double quotes to single quotes. I believe HTML uses double quotes by default, and PHP uses single quotes.

PHP can and does use both, it depends on the context. Double quotes will parse out variables, while single quotes will not. If you've ever seen spots that go `"{$variable}"`, that's part of why, though `"$variable"` will do the same thing. If $variable" is set to "mystring", both will output `"mystring"`. Single quotes would output the $ sign literally.

With that said, I chose double quotes for this because there's much more likely to be cases of apostrophes, especially around the labels, and if you try to include those in the output, they would actually make un-pasteable snippets without extra cleanup. So, I opted to go with double quotes to avoid having to escape the intended apostrophes, because `\'` in the middle of things looks not as pretty as it could be in my mind. Then the rest of the quotes went with doubles for consistency.

> Replaced the functions with closures. Closures got added in PHP 5.3 (2009, dead in 2014). If you only reference a function once then you can use a closure instead. The added benefit is not having to name a function. Naming isn't fun, and if the name isn't unique you get an error.

Honestly, leaving this one as is is probably pretty low risk to leave as is. I also know that our user base isn't always dev-heavy in knowledge. Part of why they'd be using a plugin like ours in the first place. That said though, I know that that's not the entirety of the user base either. We have some very code-savy seasoned developers who use the plugin as well, myself included on my own websites. While probably not always the case, the idea of facilitating, at least out of the box, enough code to make their registration un-hookable isn't bad. I know I've helped people juggle registration priority with the hooks we register on. If they want to move to closures on their own, more power to them, and it's a 1min change for our code. I also know I've seen cases of us being a generator, which is not an intended use-case, but one we will happily take.

> Replaced array() with []. [] got added in PHP 5.4 (2012, dead in 2015). It's just shorter, cleaner, nicer.

I've been slowly working on this topic in the files as part of a 1.7.0 release, including the related file in question https://github.com/WebDevStudios/custom-post-type-ui/commit/ffb3ae9f6501373bb4bec6e4eceeef8bedbc734f#diff-92e783f13d096fa428f22de7da330d2a but I see I did miss some spots in the output. We are requiring PHP5.6/WP5.2 for 1.7.0 so I'll just keep taking care of this one on my own.

I definitely appreciate the thoughts and feedback, with some of them already in the works, and some not so much thus far.
